Feat/member signup

YJKim1026 opened the pull request:
회원가입, 로그인 기능 추가

- signup.js 추후 보완 예정
- Refresh Token 미구현

- css는.. 없음..

YJKim1026 committed:
fix(db): Add password column to user table

YJKim1026 committed:
feat(member): Add UserEntity and UserDto, Update build.gradle

- build.gradle 수정: JWT 및 Validation 관련 의존성 추가
- UserEntity, UserDto 추가

YJKim1026 committed:
feat(member): Add UserRepository, UserService and SecurityConfig

YJKim1026 committed:
feat(member-signup): Add Add user registration feature

- Implement user registration form in HTML
- Add email and phone number duplication check logic
- Add password confirmation and validation logic
- Validate phone number format (11 digits)
- Integrate all validation and submission logic

YJKim1026 committed:
refactor(route): Change register URL to signup

- 회원가입 페이지 url '/user/signup'으로 변경
- 회원가입 관련 파일명 register -> signup 변경

YJKim1026 committed:
feat(login): Add login

YJKim1026 committed:
feat(login): 2

YJKim1026 committed:
feat(login): Add GlobalControllerAdvice for login status handling

Add a global controller advice to manage login status across all pages.
The 'isLoggedIn' attribute is now globally available in all views.

YJKim1026 committed:
feat(signup/login):

- 미완성

Johnywhisky commented on sql/project_table_ddl.sql:
```diff
@@ -53,6 +53,9 @@ CREATE TABLE users (
 	profile_img              VARCHAR(512) NULL  DEFAULT 'default_profile_img_url'
 );
 
+ALTER TABLE users
```
제일 밑 줄로 내려주세요 :)

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/MainController.java:
```diff
@@ -1,13 +1,14 @@
+
     @GetMapping({"/", ""})
-    public String index() {
+    public String index(Model model) {
```
Model model은 삭제해주세용:)

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/controller/AuthController.java:
```diff
@@ -0,0 +1,119 @@
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequestMapping("/auth")
```
rest api 규칙에 맞게 `/users/auth`로 바꿔 주세요 :)

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/controller/AuthController.java:
```diff
@@ -0,0 +1,119 @@
+     * @param email 이메일 주소
+     * @return 중복 여부 (exists: true/false)
+     */
+    @GetMapping("/check-email")
```
rest api 규칙에 맞춰서 query string로 수정할게여
예) `/check?tpye=email&val=userEmail` 식으로 수정하면 좋을 거 같아요.
그리고 핸드폰 번호의 경우에는 `/check?tpye=phone&val=userPhone` 식으로 하면 될 거 같아요.
url을 통합하고 함수도 `duplicationCheck` 등으로 수정해서 하나로 합치면 될 것 같습니다

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/controller/AuthController.java:
```diff
@@ -0,0 +1,119 @@
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        // JWT 생성
+        String accessToken = jwtUtil.generateAccessToken(loginRequestDto.getEmail());
```
AccessToken에 email 대신에 pk로 바꿔주세요 :)

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/controller/AuthController.java:
```diff
@@ -0,0 +1,119 @@
+}
+
+    // /**
+    //  * refresh 토큰.. 나중에 해.. 뒤지겟으니깐..
```
ㅜㅜ

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/controller/UserController.java:
```diff
@@ -0,0 +1,26 @@
+     * 회원가입 페이지
+     * @return 
+     */
+    @GetMapping("/user/signup")
```
@RequestMapping 사용해서 `/users` 로 바꿔주세요

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/repository/UserRepository.java:
```diff
@@ -0,0 +1,15 @@
+    Optional<UserEntity> findByEmail(String email);
+
+    // 핸드폰 번호로 사용자 검색
+    Optional<UserEntity> findByPhone(String phone);
```
존재만 확인하는 거면 `existsBy`를 써서 체크하면 좋을 것 같아여

Johnywhisky replied:
`email` 존재 유무 체크도 `existsBy`로

Johnywhisky commented on src/main/java/com/scit/letsleave/domain/user/service/UserService.java:
```diff
@@ -0,0 +1,80 @@
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
```
`Controller`도 구분 한 김에  Service도 `AuthService`로 구분 지어서 `AuthService`에서는 로그인, 회원가입만 처리하는 쪽으로 하는게 어떨까요??

Johnywhisky commented on src/main/java/com/scit/letsleave/global/jwt/JwtUtil.java:
```diff
@@ -0,0 +1,66 @@
+@Component
+public class JwtUtil {
+
+    private static final String SECRET_KEY = "your-256-bit-secret-your-256-bit-secret";
```
TODO: 추후에 환경 변수 분리 후 수정하기(윤서준)

Johnywhisky commented:
테스트 하나 둘 셋

Johnywhisky reviewed (changes requested):
수정수정수정은크리스탈수정은나오시

YJKim1026 committed:
refactor(auth): Apply multiple changes to authentication, database, and controller

- AccessToken의 subject를 email에서 pk(id)로 변경
  - JWT 생성 및 검증 로직 수정
  - CustomUserDetailsService에서 loadUserById 메서드 추가

- 중복 확인 API를 통합하여 type과 val 쿼리 스트링으로 처리하도록 변경

- DDL 파일 수정 (password 컬럼 추가)

YJKim1026 committed:
refactor(auth): Replace findBy with existsBy for email and phone checks

- UserRepository에서 findByEmail과 findByPhone을 existsByEmail과 existsByPhone으로 변경
- UserService의 isEmailExists와 isPhoneExists 메서드에서 findBy 대신 existsBy를 사용하도록 수정

YJKim1026 committed:
refactor(controller): Change path to /users

- rest api 규칙에 맞게 경로 /user에서 /users로 변경

YJKim1026 committed:
refactor(service): Split UserService into AuthService and UserService

- AuthService 생성
  - 회원가입(registerUser) 및 비밀번호 검증(checkPassword) 로직 이동
  - 로그인 및 인증 관련 로직 전담

- UserService 수정
  - 이메일 존재 여부 확인(isEmailExists), 전화번호 존재 여부 확인(isPhoneExists), 사용자 정보 조회(findByEmail) 로직 유지
  - 사용자 정보 관리 및 조회 전담

- AuthController에서 AuthService를 사용하도록 변경

YJKim1026 committed:
refactor(auth): Extend Access Token cookie expiration to 1 hour

- Access Token 쿠키의 유효시간을 기존 15분에서 1시간으로 변경

YJKim1026 committed:
feat(global): Modify GlobalControllerAdvice for user information management

- UserDto 클래스에 엔티티 -> DTO 변환 메서드(toDto) 추가
- UserService에 findById 메서드 추가 (ID 기반 사용자 조회)
- GlobalControllerAdvice 수정
  - 모든 요청에서 로그인 여부(isLoggedIn)와 사용자 정보(userInfo)를 모델에 자동 추가

Johnywhisky commented:
@YJKim1026 자바스크립트 문제 있던 부분은 다 해결된건가요?

YJKim1026 committed:
Merge branch 'main' into feat/member-signup

YJKim1026 committed:
fix(global): Reduced the fields provided by UserInfo

- Limited UserInfo to include only id, nickname, isAgreeLoc, isAgreeNewsNoti, and isAgreeMarketingNoti(UserDto/toDto)

YJKim1026 committed:
Merge branch 'main' into feat/member-signup

O0oO0Oo reviewed (commented):
💯

O0oO0Oo reviewed (approved):
💯
